src/order/locally_finite.lean

```diff
@@ -482,6 +482,154 @@ locally_finite_order.of_Icc' (α × β)
 
 end preorder
 
+/-!
+#### `with_top`, `with_bot`
+
+Adding a `⊤` to a locally finite `order_top` keeps it locally finite.
+Adding a `⊥` to a locally finite `order_bot` keeps it locally finite.
+-/
+
+section order_top
+open with_top
+variables (α) [partial_order α] [order_top α] [locally_finite_order α]
+
+local attribute [pattern] coe
+
+instance : locally_finite_order (with_top α) :=
```

Comment:
Should there be accompanying (simp)-lemmas about Icc/Ico/etc for intervals in `with_{top,bot}`?

Reply:
I assume you'll want one for each match case? That's *a lot*, but sure.

Reply:
Are those fine?